### ISSUE TITLE
Transition from Google CLA to an OpenJSF CLA

### ISSUE THREAD

**Issue by mrjoro** [ampproject/meta#60]
Per https://github.com/ampproject/meta-tsc/issues/25 and related discussions, now that ampproject has moved from Google to an OpenJSF Growth project we will be transitioning to a new CLA.  This issue tracks the work to move to the new CLA.

@tobie can you update this issue with what the agreed upon next steps for amp were?

We should make sure to include how this affects existing contributors.

/cc @ampproject/wg-infra

**Comment by rsimha** [comment]
Bumping this to check if the coast is clear to adopt CLA Assistant (https://cla-assistant.io/) for `ampproject`.

**Comment by mrjoro** [comment]
@tobie @brianwarner @jorydotcom do you know if we're able to switch to the CLA Assistant now?

**Comment by tobie** [comment]
Apologies for not following up. This is still pending board approval.

**Comment by tobie** [comment]
We just got Board approval during yesterday's CPC meeting to move ahead with our preferred solution ([Individual CLA](https://openjsf.org/wp-content/uploads/sites/84/2019/11/OpenJS-Foundation-Individual-CLA-2019-11-15.pdf), combined with CLA-Assistant). You can go ahead and deploy it.

**Comment by mrjoro** [comment]
@tobie you mentioned this is on hold for now, right?

**Comment by tobie** [comment]
No, it’s all good. You can go ahead and implement this now. It’s been approved by the Board. The wording change to the IP policy will take a bit of time to land but there’s no reason to let that slow us down.

**Comment by patrickkettner** [comment]
@mrjoro  @tobie apologize if this is the improper channel, but a question on the OpenJS CLA - the current localization work being done by a vendor is committed via a github bot. We are insuring all actual people involved in the work have signed off on the Google CLA, but the bot doesn't have the ability to sign paperwork, and therefore the PRs need to be manually over ridden as a result of the missing "author" signature. 
Is there a way on the OpenJS CLA to get a bot approve/ignored for signoffs?

**Comment by tobie** [comment]
@patrickkettner you can just add the bot to the allowlist using the admin console: https://github.com/cla-assistant/cla-assistant#can-i-allow-bot-user-contributions.

CLA-assistant hasn't been deployed yet, however. I don't know what timeline @rsimha has in mind.

**Comment by rsimha** [comment]
I'm in the process of deploying CLA-assistant across the `ampproject` organization this week.

Here is what an un-signed CLA looks like on a PR:

![image](https://user-images.githubusercontent.com/26553114/95237536-5a0b0700-07d6-11eb-90c5-17226f1bf0dc.png)

Here is what the CLA signing UI looks like (when tested with my personal fork):
![image](https://user-images.githubusercontent.com/26553114/95237689-86bf1e80-07d6-11eb-90d0-7777eaca3cab.png)

And here is what a signed CLA looks like on a PR:

![image](https://user-images.githubusercontent.com/26553114/95237968-db629980-07d6-11eb-9a0b-bb45aebff436.png)

Before we make this a blocking check, I'll be sure to configure things so that all our existing bot accounts are unblocked by the CLA check.

**Comment by tobie** [comment]
May I suggest tuning the wording a bit to use something like "agree to" rather than "sign"?

**Comment by rsimha** [comment]
> May I suggest tuning the wording a bit to use something like "agree to" rather than "sign"?

The "Please sign the CLA for..." verbiage is not part of the CLA text, it is part of the [CLA Assistant UX](https://github.com/cla-assistant/cla-assistant/blob/3bb5f954361c2d61332f154a63fcf8cd2ce52d79/src/client/controller/cla.js#L208) and cannot be changed. Also, I think "please sign the agreement" sounds better than "please agree to the agreement".

**Comment by tobie** [comment]
Oh, I thought you could tweak that. My bad. The intention was to avoid using the term “sign,” which for a lot of CLA actually means printing a pdf, signing it, uploading, etc. and can be a deterrent to people actually starting the process started. I’ll file an issue with the project directly.

**Comment by rsimha** [comment]
With https://github.com/ampproject/amp-github-apps/pull/998, the @ampproject/wg-infra working group has verified that the new CLA Assistant check can successfully be deployed and signed on a repo in the `ampproject` organization.

At this point, we are ready to turn on the check for the entire organization. Here are the next steps:

- [ ] Enable the CLA check for the entire `ampproject` org in non-blocking mode
- [ ] Update the check's allow list with all bot accounts expected to check in commits
- [ ] Check in with all repo owners if it's okay to make CLA Assistant a required + blocking check
- [ ] Disable / uninstall the Google CLA check

**Comment by tobie** [comment]
If you're relying on SAP's infrastructure, can we make sure to have a process to store backups of the data with the OpenJSF? iirc SAP provides a CSV export functionality to do so. Is there anything you could automate there?

**Comment by tobie** [comment]
> I’ll file an issue with the project directly.

Done. See: https://github.com/cla-assistant/cla-assistant/pull/627.

**Comment by rsimha** [comment]
@tobie The OpenJS CLA has been deployed across the `ampproject` org via the CLA Assistant app. It will be changed to a required (blocking) check sometime next week, after enough contributors have had a chance to sign it. After this, we can bid farewell to the Google CLA. A note to this effect has been posted to `#contributing` channel on AMP's Slack.

Meanwhile, from initial reports, several folks are reporting that signing the CLA can be extremely slow (several seconds to load the page), and sometimes fails. Did you notice this when the initial recommendation was made to use CLA Assistant (https://github.com/ampproject/meta-tsc/issues/25#issuecomment-639667831)?

**Comment by tobie** [comment]
No. I didn’t see anything of that nature. I did notice certain imperfections in the UI related to the weird frontend tech choice for this use case, but nothing of the nature you describe.

Not sure were SAP deployed their solution, but since it’s a European company, I wouldn’t be surprised if it was based in Europe, which would explain some of the network issues. 

That said, I assumed we’d be deploying our own solution, not relying on SAP’s.

**Comment by tobie** [comment]
@rsimha No, I didn’t see anything of that nature. I did notice certain imperfections in the UI related to the weird frontend tech choice for this use case, which created slight loading delays as a result, but nothing of the nature you describe.

Not sure where SAP deployed their solution, but since it’s a European company, I wouldn’t be surprised if it was deployed in Europe, which would explain some of the network issues would be worse on your side.

That said, I was working under the assumption that we would be deploying our own solution, not relying on SAP’s.

**Comment by rsimha** [comment]
@tobie, thanks for the note.

Until now, the requirement as I understood it was for us to install and set up the existing [CLA Assistant](https://cla-assistant.io/) app for the `ampproject` org. I now see that deploying our own instance is an option as well. While this may end up being the desired path forward, it will involve a significantly larger engineering and maintenance burden on our part in the long term. I'll discuss this offline with @mrjoro and post an update here. Meanwhile, the currently deployed instance of the app (cla-assistant.io) appears to be working well after some initial sign-up slowness.

Meanwhile, I have a separate unrelated question about the OpenJS CLA. A subset of the PRs merged to the `amphtml` repo by @ampproject/wg-caching involve [syncing commits](https://github.com/ampproject/amphtml/pulls?q=is%3Apr+sort%3Aupdated-desc+%22Sync+from+Google%22) from a Google-internal codebase. While these PRs are created by a member of the AMP team who has signed the CLA, they may contain commits originally authored by someone outside the AMP team. Do these individuals also need to sign the CLA? Or is it sufficient if the PR creator has signed? (Let me know if you'd like me to file a separate issue for this discussion.)

**Comment by tobie** [comment]
Thanks for the update, @rsimha. Would you mind including me in that meeting? I think it would be easier to clarify all of this in person.

**Comment by rsimha** [comment]
Update: The OpenJS CLA is now a required check for all repos in the `ampproject` org that receive code commits. All integrations with the Google CLA have been removed, and it is no longer a required check.
